gcp/modules/couchdb/main.tf

```diff
@@ -142,15 +144,41 @@ resource "null_resource" "couchdb_enable_pv_backups" {
 }
 
 resource "null_resource" "couchdb_destroy_pvcs" {
+  count = "${var.execute_destroy_pvcs == "true" ? 1 : 0}"
+
   provisioner "local-exec" {
     when = "destroy"
 
     command = <<EOF
-      if [ "${var.execute_destroy_pvcs}" == "true" ]; then
-        for PVC in $(kubectl get pvc --namespace ${var.release_namespace} -o json | jq --raw-output '.items[] | select(.metadata.name | startswith("database-storage-couchdb")) | .metadata.name'); do
-          timeout -t 600 kubectl --namespace ${var.release_namespace} delete --ignore-not-found --grace-period=300 pvc $PVC
+      for PVC in "$(kubectl get pvc --namespace ${var.release_namespace} -o json | jq -r '.items[] | select(.metadata.name | startswith("database-storage-couchdb")) | .metadata.name')"; do
+        timeout -t 600 kubectl -n ${var.release_namespace} delete --ignore-not-found --grace-period=300 pvc $PVC
+      done
+      for PV in "$(kubectl get pv -o json | jq -r '.items[] | select (.spec.claimRef.name | startswith("database-storage-couchdb")) | select(.spec.claimRef.namespace == "${var.release_namespace}") | .metadata.name')"; do
+        timeout -t 600 kubectl delete --ignore-not-found --grace-period=300 pv $PV
+      done
+      # Some times GCP disks persist even after PVC and PVs destruction, removing them with gcloud to not leave stale resources behind
+      # More info: https://issues.gpii.net/browse/GPII-3743
+      RETRIES=10
+      RETRY_COUNT=1
+      while [ "$DISKS_DELETED" != "true" ]; do
+        DISKS_DELETED="true"
+        echo "[Try $RETRY_COUNT of $RETRIES] Deleting CouchDB disks..."
+        for DISK in $(gcloud compute disks list --filter description:couchdb --format json | jq -c '.[]'); do
```

Comment:
It will not work with quotes:
```bash
# for i in "1 2 3"; do echo $i; echo "check"; done
1 2 3
check

# for i in 1 2 3; do echo $i; echo "check"; done
1
check
2
check
3
check
```

Reply:
I don't think I understand your example. I tried number of scenarios with unexpected input from `gcloud` – when `jq` can not extract expected fields from json, it leaves `stdout` empty, so execution flow never reaches the loop body.